lib/ViewModels/DataCatalogTabViewModel.js

```diff
@@ -38,4 +40,32 @@ DataCatalogTabViewModel.prototype.showInfo = function(item) {
     CatalogItemInfoViewModel.open('ui', item);
 };
 
+DataCatalogTabViewModel.prototype.clickEnabled = function(item) {
+    ga('send', 'event', 'dataSource', 'clickItem', item.name);
+
+    if (!item.isEnabled && defined(item.initialMessage) && item.initialMessage.length > 0) {
+        PopupMessageViewModel.open('ui', {
+            title: 'Data Set Information',
+            message: '<div>'+ item.initialMessage +'</div>'
+        });
+        item.initialMessage = undefined;
+    }
+
+    item.toggleEnabled();
+};
+
+DataCatalogTabViewModel.prototype.clickOpen = function(item) {
+    ga('send', 'event', 'dataSource', 'clickOpen', item.name);
+
+    if (!item.isOpen && defined(item.initialMessage) && item.initialMessage.length > 0) {
+        PopupMessageViewModel.open('ui', {
+            title: 'Data Provider Information',
+            message: '<div>'+ item.initialMessage +'</div>'
+        });
+        item.initialMessage = undefined;
```

Comment:
Ok, but that's just one example of why losing information like this isn't a good idea, especially since it's easy to fix.